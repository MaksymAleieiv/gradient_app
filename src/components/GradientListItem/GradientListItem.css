

.gradient-block {
    width: 100%;
    height: 90px;
    background-size: cover;
    border-radius: var(--border-rad);
    display: flex;
    border: 1px solid var(--main-color);
    margin-bottom: 12px;
}

.gradient-block__colors {
    width: calc(100% - 48px);
    border-right: 1px solid var(--main-color);
    border-top-left-radius: var(--border-rad);
    border-bottom-left-radius: var(--border-rad);
    padding: 0 16px;
    display: flex;
    justify-content: space-between;
    align-items: center;
}

.gradient-block__colors > span {
    font-size: 24px;
    color: var(--whitey);
}

.gradient-block__buttons {
    width: 48px;
    border-top-right-radius: var(--border-rad);
    border-bottom-right-radius: var(--border-rad);
}

.gradient-block__buttons > * {
    width: 100%;
    height: 50%;
    border: none;
    cursor: pointer;
    background-color: var(--whitey);
    background-repeat: no-repeat;
    background-position: center;

}

.gradient-block__buttons__remove {
    border-bottom: 1px solid var(--main-color);
    background-image: url(../../assets/delete-icon.png);
    background-size: 26px;
    border-top-right-radius: var(--border-rad);
}

.gradient-block__buttons__edit {
    background-image: url(../../assets/edit-icon.png);
    background-size: 32px;
    display: block;
    border-bottom-right-radius: var(--border-rad);
}