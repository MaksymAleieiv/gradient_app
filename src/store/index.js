import { createStore } from 'redux'
import { gradientsList_Reducer as gradientsList } from './reducers/gradients-list/gradientsList_Reducer'
import { persistStore, persistReducer } from 'redux-persist'
import storage from 'redux-persist/lib/storage'

const persistConfig = {
    key: 'root',
    storage
}

const persistedReducer = persistReducer(persistConfig, gradientsList)


let store = createStore(persistedReducer)
let persistor = persistStore(store)
export { store, persistor }
