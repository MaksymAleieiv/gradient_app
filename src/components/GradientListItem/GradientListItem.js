import React from 'react'
import { Link } from 'react-router-dom'
import { useDispatch } from 'react-redux'
import { gradientsList_ActionCreators } from '../../store/reducers/gradients-list/gradientsList_ActionCreators'
import './GradientListItem.css'

function GradientListItem({gradient}) {

    const dispatch = useDispatch()

    const removeThisItem = () => {
        dispatch(gradientsList_ActionCreators.removeGradient(gradient))
    }

    return (
        <div style={{background: `linear-gradient(to right, ${gradient.firstColor}, ${gradient.secondColor})`}} className='gradient-block'>
            <div className='gradient-block__colors'>
                <span>{gradient.firstColor}</span>
                <span>{gradient.secondColor}</span>
            </div>
            <div className='gradient-block__buttons'>
                <button className='gradient-block__buttons__remove' onClick={removeThisItem}>Â </button>
                <Link className='gradient-block__buttons__edit' to={`/edit/${gradient.id}`}/>
            </div>
        </div>
    )
}

export default React.memo(GradientListItem)
