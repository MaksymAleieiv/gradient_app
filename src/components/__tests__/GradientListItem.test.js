import { cleanup } from '@testing-library/react';
import {BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer'
import GradientListItem from '../GradientListItem/GradientListItem';

import { Provider } from 'react-redux';
import { store } from '../../store/index'

afterEach(() => {
    cleanup();
})

test('gradient list item snapshot, i want to die', () => {
    const gradient = { id: '10', firstColor: '#fff', secondColor: '#000' };
    const tree = renderer.create(
        <Provider store={store}>
            <Router>
                <GradientListItem gradient={gradient}/>
            </Router>
        </Provider>
            ).toJSON();
    expect(tree).toMatchSnapshot();
})