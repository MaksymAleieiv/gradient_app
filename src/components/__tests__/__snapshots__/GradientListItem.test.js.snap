// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gradient list item snapshot, i want to die 1`] = `
<div
  className="gradient-block"
  style={
    Object {
      "background": "linear-gradient(to right, #fff, #000)",
    }
  }
>
  <div
    className="gradient-block__colors"
  >
    <span>
      #fff
    </span>
    <span>
      #000
    </span>
  </div>
  <div
    className="gradient-block__buttons"
  >
    <button
      className="gradient-block__buttons__remove"
      onClick={[Function]}
    >
      Â 
    </button>
    <a
      className="gradient-block__buttons__edit"
      href="/edit/10"
      onClick={[Function]}
    />
  </div>
</div>
`;
